File: programmers/완주하지_못한_선수/완주하지_못한_선수.js
function solution(participant, completion) {
  const stock = new Map();
  participant.forEach((player) => {
      const val = stock.get(player);
      if(val) stock.set(player, val+1);
      else stock.set(player, 1);
  });

  completion.forEach((player) => {
      const val = stock.get(player);
      stock.set(player, val-1);
  });

  let answer;
  stock.forEach((v, k, map) => {if(v === 1) answer = k;});
  return answer;
}

File: programmers/완주하지_못한_선수/완주하지_못한_선수.java
import java.util.stream.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        participant = Stream.of(participant).sorted().toArray(String[]::new);
        completion = Stream.of(completion).sorted().toArray(String[]::new);

        int i;
        for(i = 0 ; i < completion.length ; i++ ){
            if(!completion[i].equals(participant[i])) return participant[i];
        }

        return participant[i];
    }
}
